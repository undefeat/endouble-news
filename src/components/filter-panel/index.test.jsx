import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import shallow from 'react-test-renderer/shallow';
import FilterPanel from '.';

const countries = [
    { code: 'us', name: 'United State', nativeName: 'United States of America' },
    { code: 'nl', name: 'Netherlands', nativeName: 'Nederland' },
    { code: 'uk', name: 'United Kingdom', nativeName: 'United Kingdom' },
];
const categories = ['general', 'sports', 'buisness'];

describe('FilterPanel', () => {
    it('should correctly render', () => {
        const tree = shallow
            .createRenderer()
            .render(
                <FilterPanel
                    q="test"
                    selectedCategory="sports"
                    selectedCountry="nl"
                    categories={categories}
                    countries={countries}
                    onCategoryChanged={() => {}}
                    onCountryChanged={() => {}}
                    onSearchPhraseChanged={() => {}}
                />,
            );

        expect(tree).toMatchSnapshot();
    });

    it('should call change callbacks when input changes', () => {
        const onCategoryChanged = jest.fn();
        const onCountryChanged = jest.fn();
        const onSearchPhraseChanged = jest.fn();
        const categoryChangeEvent = { target: { value: 'new category value' } };
        const countryChangeEvent = { target: { value: 'new country value' } };

        const element = ReactTestUtils.renderIntoDocument(
            <FilterPanel
                q="test"
                selectedCategory="sports"
                selectedCountry="nl"
                categories={categories}
                countries={countries}
                onCategoryChanged={onCategoryChanged}
                onCountryChanged={onCountryChanged}
                onSearchPhraseChanged={onSearchPhraseChanged}
            />,
        );

        const searchInput = ReactTestUtils.findRenderedDOMComponentWithClass(element, 'filter-panel__input');
        const countrySelect = ReactTestUtils.findRenderedDOMComponentWithClass(
            element,
            'filter-panel__select--country',
        );
        const categorySelect = ReactTestUtils.findRenderedDOMComponentWithClass(
            element,
            'filter-panel__select--category',
        );

        ReactTestUtils.Simulate.change(searchInput);
        expect(onSearchPhraseChanged).not.toHaveBeenCalled(); // Because we use debounce.

        ReactTestUtils.Simulate.change(countrySelect, countryChangeEvent);
        expect(onCountryChanged).toHaveBeenCalledTimes(1);
        expect(onCountryChanged).toHaveBeenCalledWith(countryChangeEvent.target.value);

        ReactTestUtils.Simulate.change(categorySelect, categoryChangeEvent);
        expect(onCategoryChanged).toHaveBeenCalledTimes(1);
        expect(onCategoryChanged).toHaveBeenCalledWith(categoryChangeEvent.target.value);
    });
});
