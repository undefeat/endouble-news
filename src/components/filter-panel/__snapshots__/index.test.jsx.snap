// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterPanel should correctly render 1`] = `
<div
  className="filter-panel"
>
  <input
    aria-label="Keywords or a phrase to search for"
    className="filter-panel__input"
    onChange={[Function]}
    placeholder="Search..."
    type="search"
    value="test"
  />
  <label
    className="filter-panel__label"
    htmlFor="country-select"
  >
    <span>
      Country:
    </span>
    <select
      className="filter-panel__select filter-panel__country-select"
      id="country-select"
      onChange={[Function]}
      value="nl"
    >
      <option
        value="us"
      >
        United States of America
      </option>
      <option
        value="nl"
      >
        Nederland
      </option>
      <option
        value="uk"
      >
        United Kingdom
      </option>
    </select>
  </label>
  <label
    className="filter-panel__label"
    htmlFor="category-select"
  >
    <span>
      Category:
    </span>
    <select
      className="filter-panel__select filter-panel__category-select"
      id="category-select"
      onChange={[Function]}
      value="sports"
    >
      <option
        value="general"
      >
        general
      </option>
      <option
        value="sports"
      >
        sports
      </option>
      <option
        value="buisness"
      >
        buisness
      </option>
    </select>
  </label>
</div>
`;
